src/mame/seibu/sei021x_sei0220_spr.h

```diff
@@ -0,0 +1,80 @@
+// license:BSD-3-Clause
+// copyright-holders:David Graves, Angelo Salese, David Haywood, Tomasz Slanina, Carlos A. Lozano, Bryan McPhail, Pierpaolo Prazzoli
+#ifndef MAME_VIDEO_SEI021X_SEI0220_SPR_H
+#define MAME_VIDEO_SEI021X_SEI0220_SPR_H
+
+#pragma once
+
+class sei0210_device : public device_t, public device_gfx_interface
+{
+public:
+	typedef device_delegate<u32 (u8 pri, u8 ext)> pri_cb_delegate;
+	typedef device_delegate<u32 (u32 code, u8 ext, u8 y)> gfxbank_cb_delegate;
+
+	sei0210_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock);
+	template <typename T> sei0210_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock, T &&palette_tag, const gfx_decode_entry *gfxinfo)
+		: sei0210_device(mconfig, tag, owner, clock)
+	{
+		set_info(gfxinfo);
+		set_palette(std::forward<T>(palette_tag));
+	}
+
+	// configuration
+	template <typename... T> void set_pri_callback(T &&... args) { m_pri_cb.set(std::forward<T>(args)...); }
+	template <typename... T> void set_gfxbank_callback(T &&... args) { m_gfxbank_cb.set(std::forward<T>(args)...); }
+	void set_alt_format(bool alt_format) { m_alt_format = alt_format; }
+	void set_offset(s32 xoffset, s32 yoffset)
+	{
+		m_xoffset = xoffset;
+		m_yoffset = yoffset;
+	}
+
+	void draw_sprites(screen_device &screen, bitmap_ind16 &bitmap, const rectangle cliprect, u16* spriteram, u16 size);
+	void draw_sprites(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle cliprect, u16* spriteram, u16 size);
+
+protected:
+	sei0210_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, u32 clock);
+
+	virtual void device_start() override;
+	virtual void device_reset() override;
+
+	virtual s32 get_coordinate(s32 coordinate)
+	{
+		return (coordinate & 0x8000) ? (0 - (0x200 - (coordinate & 0x1ff))) : (coordinate & 0x1ff);
+	}
+
+private:
+	template<class T>
+	void draw(screen_device &screen, T &bitmap, const rectangle cliprect, u16* spriteram, u16 size);
+
+	pri_cb_delegate     m_pri_cb;
+	gfxbank_cb_delegate m_gfxbank_cb;
+
+	bool m_alt_format = false;
+	s32 m_xoffset = 0;
+	s32 m_yoffset = 0;
```

Comment:
Please don’t put member initialisation in class declarations when the constructor is in a different file.